pinot-segment-spi/src/main/java/org/apache/pinot/segment/spi/index/reader/ForwardIndexReader.java

```diff
@@ -258,6 +259,45 @@ default void readValuesSV(int[] docIds, int length, double[] values, T context)
     }
   }
 
+  /**
+   * Fills the values
+   * @param docIds Array containing the document ids to read
+   * @param length Number of values to read
+   * @param values Values to fill
+   * @param context Reader context
+   */
+  default void readValuesSV(int[] docIds, int length, BigDecimal[] values, T context) {
```

Comment:
We should add raw index support to the `BIG_DECIMAL` type (we may add a todo here and address it in a separate PR. It is important though because in most of the cases it will be more efficient to store big decimal as raw)

Reply:
@Jackie-Jiang added todo. Will address in next PR.